src/components/button/button-types.style.js

```diff
@@ -0,0 +1,62 @@
+export default ({ colors }) => ({
```

Comment:
why not just return a string, that approach will save us a lot of unnecessary logic
```
primary: `
  background: ${colors.primary},
  color: ${colors.white}
  &:hover {
      background: ${colors.secondary},
      color: ${colors.white}
  }
`,
secondary: `
.
.
.
```

Reply:
There's still the issue of having to then define the `disabled` style for each state the button is in, as it looks different when it is `primary` than when it is `secondary` etc